cmd/src/actions_exec_backend_runner.go

```diff
@@ -363,3 +359,62 @@ func repositoryZipArchiveURL(repoName, rev, token string) (*url.URL, error) {
 	u.Path = path.Join(u.Path, repoName+"@"+rev, "-", "raw")
 	return u, nil
 }
+
+func unzip(zipFile, dest string) error {
+	r, err := zip.OpenReader(zipFile)
+	if err != nil {
+		return err
+	}
+	defer r.Close()
+
+	for _, f := range r.File {
+		fpath := filepath.Join(dest, f.Name)
+
+		// Check for ZipSlip. More Info: http://bit.ly/2MsjAWE
+		if !strings.HasPrefix(fpath, filepath.Clean(dest)+string(os.PathSeparator)) {
```

Comment:
I'd advise hoisting the prefix assembly out of the loop:

```go
outputBase := filepath.Clean(dest)+string(os.PathSeparator)
// …
if !strings.HasPrefix(fpath, outputBase) {
```

Not so much for performance but to be self-documenting.